### PR TITLE
Make events + fix things

### LINKED ISSUE #45: Events/Posts - Frontend
### **Acceptance Criteria**
BLOCKED BY: https://github.com/xusandrew/cs-348-project/issues/49

**Create Events**
- [ ] Add a create events page with a form element that takes in the following info: title, description, location, start_time, end_time
   - Ensure that only club admins can view this page
- [ ] Add typeguards/checks to make sure user entered data is appropriate (i.e. we cannot have unnamed events)
- [ ] Send the following event data to the backend:
```
  cid: number
  title: string //from user input
  description: string | null  //from user input
  location: string | null  //from user input
  start_time: Date | null  //from user input
  end_time: Date | null //from user input
  posted_time: Date
  club: Club
```

**Update Events**
- [ ] On the clubs page that is only visible to club admins, add a button on each of the existing events components that allows club admins to edit the specific event
- [ ] Add a module/page with a form element that allows the user to edit the information of a specific event
- [ ] Send the following data to the backend:
```
  eid: number 
  cid: number
  title: string //from user input
  description: string | null  //from user input
  location: string | null  //from user input
  start_time: Date | null  //from user input
  end_time: Date | null //from user input
  posted_time: Date
  club: Club
```

**Delete Events**
- [ ] On the clubs page that is only visible to club admins, add a button on each of the existing events components that allows club admins to delete the specific event
- [ ] [Optional] Add a confirmation module for deleting the event
- [ ] Send the following data to the backend:
```
  eid: number 
```

### PR DESCRIPTION
**Pull request by xusandrew**
big diff is from generating package-lock.json for frontend and email starter

also reformatted some files

### CODE REVIEW

File: frontend/data/components/login.tsx
Comment (xusandrew) [inline frontend/data/components/login.tsx:1]: only removed this comment

File: frontend/src/app/email/RsvpSignUpEamil.tsx
Comment (xusandrew) [inline frontend/src/app/email/RsvpSignUpEamil.tsx:1]: only formatting

File: frontend/src/app/email/getTemplate.tsx
Comment (xusandrew) [inline frontend/src/app/email/getTemplate.tsx:1]: only formatting

File: frontend/src/components/events/Rsvp.tsx
Comment (xusandrew) [inline frontend/src/components/events/Rsvp.tsx:1]: just formatting

**Review by SimonZhang04 — APPROVED** [review]
k

**Pull request merged**